cmd/stack.go

```diff
@@ -332,3 +350,22 @@ func printStatus(cmd *cobra.Command, servicesStatus []stack.ServiceStatus) {
 	t.SetStyle(table.StyleRounded)
 	cmd.Println(t.Render())
 }
+
+func detectShell() (string, error) {
+	ppid := os.Getppid()
+
+	parent, err := sysinfo.Process(ppid)
+	if err != nil {
+		return "", fmt.Errorf("cannot retrieve information for process %d: %w", ppid, err)
+	}
+
+	parentInfo, err := parent.Info()
+	if err != nil {
+		return "", fmt.Errorf("cannot retrieve information for parent of process %d: %w", ppid, err)
+	}
+
+	shell := filepath.Base(parentInfo.Exe)
```

Comment:
Found another corner case.

When running `elastic-package` with `go run`, it detects `go` as the shell:
```
detected shell: go
Error: shellinit failed: cannot get shell init template: shell type is unknown, should be one of bash, fish, sh, zsh
exit status 1
```

This happens when running `elastic-package` with `go run` in repositories where this package is in `go.mod`. This is done in CI in some cases, and for example I use it when trying to reproduce failures in the integrations repository with the same version of elastic-package that is used in CI.

This could be considered a regression, could we do something about this? Maybe it is possible to detect if `elastic-package` is being run with `go run`, or we could get the parent of the parent when it is `go`.

Reply:
Or maybe another option to avoid regressions, we could directly return `bash` and log something to stderr instead of failing when this method cannot detect the shell.

Reply:
This use case is supported by not using auto detection (passing `--shell=bash`). I think that is preferable than deciding to fallback to BASH because that can lead to errors that are very hard to troubleshoot and less explanatory.

We may want to make this explicit using a different error message.

Reply:
Well, let's separate my two comments:
* Not breaking the not so rare `go run` case. Possible workaround for this could be: if found shell is `go`, try to get the parent of the parent instead.
* General solution to don't break unknown cases that work now. I propose to use `bash` in case of error only to keep the current behavior, where only bash is supported. I agree that it could be more discussed as I don't have more use cases beyond the mentioned ones, and it would be better to have explicit behavior.

Reply:
> Not breaking the not so rare go run case. Possible workaround for this could be: if found shell is go, try to get the parent of the parent instead.

I added this in https://github.com/elastic/elastic-package/pull/921/commits/12ddc3d18b9c71a8fd7b1fa72717a5ee4e5cecd4 (with tests)

> General solution to don't break unknown cases that work now. I propose to use bash in case of error only to keep the current behavior, where only bash is supported.

I didn't consider this POV, but it make complete sense, as the behaviour would not change from current implementation. I will change the default case.